### DISCUSSION TITLE
How to use `failureCondition` / `successCondition` to check if metric increased compared to previous after deployment

### DISCUSSION THREAD

**Discussion by Dentrax** [argoproj/argo-rollouts#2106] [General]
Hey,

I have read the [Analysis](https://argoproj.github.io/argo-rollouts/features/analysis/) docs, but most of the `failureCondition` and `successCondition` examples uses conditions like `result[0] >= 7` or `result == 'true'`. What I want to achieve is that something like:

I have a Prometheus metric called: `myapp_critical_errors`. Let's assume its current value at time _t_ is `7`. After I roll out the app, I expect this value not to increase. So this is where I have to check previous value:

The following case means something gone wrong in the deployment and Argo CD should roll back.
```
myapp_critical_errors[t+1] > myapp_critical_errors[t]
```

I have to tell to Argo CD something like: _"in current deployment, myapp_critical_errors metric's value is 7, if this value increases after next deployment, you have to roll back the deployment"_

Q: How can I create the conditions in `AnalysisTemplate` to reproduce the following above case?

**Comment by kostis-codefresh** [comment]
I think this is more a Prometheus question instead of Rollouts question.

why not use the rate/increase/delta functions of prometheus ? This way you check if the rate is 0 (no new errors were added) or > 1 (new errors were present).
